skrifa/src/color/traversal_tests.rs

```diff
@@ -0,0 +1,341 @@
+            paint_dump: PaintDump {
+                glyph_id: gid,
+                ..Default::default()
+            },
+        }
+    }
+
+    pub fn into_inner(self) -> PaintDump {
+        self.paint_dump
+    }
+}
+
+fn location_to_filename<I>(set_name: &str, settings: I) -> String
+where
+    I: IntoIterator,
+    I::Item: Into<VariationSetting>,
+{
+    let formatted_settings: Vec<String> = settings
+        .into_iter()
+        .map(|entry| {
+            let entry_setting = entry.into();
+            format!("{:}_{:}", entry_setting.selector, entry_setting.value)
+        })
+        .collect();
+    let suffix = match formatted_settings.len() {
+        0 => String::new(),
+        _ => format!("_{}", formatted_settings.join("_")),
+    };
+    format!("colrv1_{}{}", set_name.to_lowercase(), suffix)
+}
+
+fn colrv1_traversal_test(set_name: &str, test_chars: &[char], settings: &[(&str, f32)]) {
+    let colr_font = font_test_data::COLRV0V1_VARIABLE;
+    let font = FontRef::new(colr_font).unwrap();
+
+    let location = font.axes().location(settings);
+
+    let dumpfile_path = format!(
+        "../font-test-data/test_data/colrv1_json/{}",
+        location_to_filename(set_name, settings)
+    );
+
+    let test_gids = test_chars
+        .iter()
+        .map(|codepoint| font.charmap().map(*codepoint).unwrap());
+
+    let paint_dumps_iter = test_gids.map(|gid| {
+        let mut color_painter = JsonDumpColorPainter::new(gid.to_u16());
+        traverse_color_outline(&font, gid, (&location).into(), &mut color_painter).unwrap();
+        color_painter.into_inner()
+    });
+
+    static GENERATE: bool = false;
+    if GENERATE {
+        let mut file = OpenOptions::new()
+            .write(true)
+            .append(true)
+            .create(true)
+            .open(dumpfile_path)
+            .unwrap();
+
+        paint_dumps_iter.for_each(|dump| {
+            writeln!(file, "{}", serde_json::to_string(&dump).unwrap())
+                .expect("Writing to file failed.")
+        });
+    } else {
+        let file = File::open(dumpfile_path).expect("Unable to open expectations file.");
+        let mut lines = io::BufReader::new(file).lines();
+        for dump in paint_dumps_iter {
+            match lines.next() {
+                Some(line) => {
+                    assert!(
+                        dump == serde_json::from_str(
+                            &line.as_ref().expect("Failed to read expectations line from file.")
+                        )
+                        .expect("Failed to parse expectations line."),
+                        "Result did not match expectation for glyph id: {}\nActual: {}\nExpected: {}\n",
+                        dump.glyph_id, serde_json::to_string(&dump).unwrap(), &line.unwrap()
+                    )
+                }
+                None => panic!("Expectation not found for glyph id: {}", dump.glyph_id),
+            }
+        }
+    }
+}
+
+macro_rules! make_ident_string {
```

Comment:
Any reason for this macro rather than calling stringify! directly in the one below?

Reply:
At some point I just go `$glyph_set` in the output then - but trying this again, now it worked. 🤷